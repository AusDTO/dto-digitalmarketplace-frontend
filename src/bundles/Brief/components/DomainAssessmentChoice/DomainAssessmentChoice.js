import React from 'react';
import {
    LocalForm,
    Field
} from 'react-redux-form';

import domains from '../../../SellerRegistration/components/DomainSelector/domains';
import Layout        from '../../../../shared/Layout';
import find from 'lodash/find'
export class DomainAssessmentChoice extends React.Component {

    state = {
        error: false
    };

    handleSubmit(val) {
        var domain = val['domain'];
        if (document.querySelector('input[name="domain"]:checked')) {
            domain = document.querySelector('input[name="domain"]:checked').value
        }
        if (domain) {
            window.location.href = this.props.assessmentUrl + "/" + domain;
        } else {
            this.setState({
                error: true
            });
            return false;
        }
    }

    render() {
        console.log({this})
        let props = this.props;
        return (
            <Layout>
                <header>
                    <h1>Choose an area of expertise for assessment</h1>
                </header>
                <article role="main">
                    <p>To respond to an outcome opportunity, you need an assessed area of expertise. To add a different area of expertise, <a href="/sellers/edit">update your seller
                            profile.</a></p>
                    <LocalForm
                        action={props.assessmentUrl}
                        onSubmit={(val) => this.handleSubmit(val)}
                    >
                        <fieldset className="field" >
                            <legend> Please choose from the following areas: </legend>
                            {this.state.error && <a className="validation-message" href="#domain">
                                <span className="visuallyhidden">Validation Error: </span>
                                <span>Please pick one area of expertise</span>
                            </a>}
                            <Field id="domain" model=".domain" dynamic={false}>

                                {Object.keys(props.domains).map((key, i) => {

                                    let fieldId = `domain-${key}`;
                                    return (
                                        <span key={i}>
                                <input type="radio"
                                       name="domain"
                                       id={fieldId}
                                       value={key}
                                />
                                <label htmlFor={fieldId}>{props.domains[key]} <p>{
                                    find(domains, {'label':  props.domains[key]}) ?
                                        find(domains, {'label':  props.domains[key]})['description'] : ""

                                }</p></label>
                            </span>
                                    )
                                })}
                            </Field>
                        </fieldset>
                        <button type="submit">Request an assessment</button>
                    </LocalForm>
                </article>
            </Layout>
        )
    }
}

DomainAssessmentChoice.propTypes = {
    domains: React.PropTypes.object,
    brief_id: React.PropTypes.number,
    assessmentUrl: React.PropTypes.string
}

export default DomainAssessmentChoice;
