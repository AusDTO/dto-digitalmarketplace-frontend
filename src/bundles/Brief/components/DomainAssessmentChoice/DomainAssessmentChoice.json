{
  "domains": {"1":"Strategy and Policy", "13":"Emerging technologies"}
}