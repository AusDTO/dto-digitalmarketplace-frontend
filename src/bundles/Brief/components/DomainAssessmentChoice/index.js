import DomainAssessmentChoice from './DomainAssessmentChoice';
export default DomainAssessmentChoice;