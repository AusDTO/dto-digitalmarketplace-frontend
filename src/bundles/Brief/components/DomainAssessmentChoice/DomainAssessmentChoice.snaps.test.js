// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import renderer from 'react-test-renderer';

import DomainAssessmentChoice from './DomainAssessmentChoice';
import sampleState from './DomainAssessmentChoice.json';

test('DomainAssessmentChoice renders', () => {
  const component = renderer.create(
        <DomainAssessmentChoice {...sampleState} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});