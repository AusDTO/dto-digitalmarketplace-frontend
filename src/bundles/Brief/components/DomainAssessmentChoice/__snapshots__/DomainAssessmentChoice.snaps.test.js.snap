// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainAssessmentChoice renders 1`] = `
<div
  className="layout"
>
  <header>
    <h1>
      Choose an area of expertise for assessment
    </h1>
  </header>
  <article
    role="main"
  >
    <p>
      To respond to an outcome opportunity, you need an assessed area of expertise. To add a different area of expertise, 
      <a
        href="/sellers/edit"
      >
        update your seller profile.
      </a>
    </p>
    <form
      action={undefined}
      noValidate={undefined}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <fieldset
        className="field"
      >
        <legend>
           Please choose from the following areas: 
        </legend>
        <div
          id="domain"
        >
          <span>
            <input
              checked={false}
              defaultChecked={undefined}
              defaultValue={undefined}
              disabled={false}
              id="domain-1"
              name="domain"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              type="radio"
              value="1"
            />
            <label
              htmlFor="domain-1"
            >
              Strategy and Policy
               
              <p>
                Strategy and policy relates to the higher level planning and policy considerations of a digital product or service. Examples include business case development, policy analysis and policy advice.
              </p>
            </label>
          </span>
          <span>
            <input
              checked={false}
              defaultChecked={undefined}
              defaultValue={undefined}
              disabled={false}
              id="domain-13"
              name="domain"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              type="radio"
              value="13"
            />
            <label
              htmlFor="domain-13"
            >
              Emerging technologies
               
              <p>
                Emerging technology is about developing proof of concepts using recent digital advances for commercial or government purposes. Examples include artificial intelligence, robotics, wearables, Internet of Things and augmented or virtual reality.
              </p>
            </label>
          </span>
        </div>
      </fieldset>
      <button
        type="submit"
      >
        Request an assessment
      </button>
    </form>
  </article>
</div>
`;
